agents/agents-utils/src/commonMain/kotlin/ai/koog/agents/utils/ModelInfo.kt

```diff
@@ -0,0 +1,56 @@
+package ai.koog.agents.utils
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Represents model information for agent events with a focus on essential data.
+ * 
+ * This data class provides the core model identification needed for agent events,
+ * including key model characteristics that are useful for debugging and monitoring.
+ * It serves as a lightweight representation optimized for serialization and backwards compatibility.
+ *
+ * @property provider The provider identifier (e.g., "openai", "google", "anthropic")
+ * @property model The model identifier (e.g., "gpt-4", "claude-3")
+ * @property displayName Optional human-readable display name for the model
+ * @property contextLength Maximum number of tokens the model can process
+ * @property maxOutputTokens Maximum number of tokens the model can generate
+ */
+@Serializable
+public data class ModelInfo(
+    val provider: String,
+    val model: String,
+    val displayName: String? = null,
+    val contextLength: Long? = null,
+    val maxOutputTokens: Long? = null
+) {
+    /**
+     * Backwards compatibility: provides the same string format as current eventString
+     * Format: "provider:model"
+     */
+    public val eventString: String get() = "$provider:$model"
```

Comment:
Let's deprecate this as well. I think we need to drop it in the next release in favour of a new format string: `provider/model_id` below.